shared/js/background/privacy-practices.es6.js

```diff
@@ -0,0 +1,115 @@
+                    score = 0
+                } else if (tosdrClass === 'B') {
+                    score = 1
+                } else if (tosdrClass === 'D' || tosdrScore > 150) {
+                    score = 10
+                } else if (tosdrClass === 'C' || tosdrScore > 100) {
+                    score = 7
+                }
+
+                tosdrScores[site] = score
+
+                // if the site has a parent entity, propagate the score to that, too
+                // but only if the score is higher
+                //
+                // basically, a parent entity's privacy score is as bad as
+                // that of the worst site it owns
+                const parentEntity = utils.findParent(site)
+
+                if (parentEntity && (!tosdrScores[parentEntity] || tosdrScores[parentEntity] < score)) {
+                    tosdrScores[parentEntity] = score
+                }
+            }
+        })
+    }
+
+    getTosdr (url) {
+        let domain = tldjs.getDomain(url)
+        let tosdrData
+
+        tosdrRegexList.some(tosdrSite => {
+            let match = tosdrSite.exec(domain)
+
+            if (!match) return
+
+            tosdrData = tosdr[match[0]]
+
+            return tosdrData
+        })
+
+        if (!tosdrData) return {}
+
+        const matchGood = (tosdrData.match && tosdrData.match.good) || []
+        const matchBad = (tosdrData.match && tosdrData.match.bad) || []
+
+        // tosdr message
+        // 1. If we have a defined tosdr class look up the message in constants
+        //    for the corresponding letter class
+        // 2. If there are both good and bad points -> 'mixed'
+        // 3. Else use the calculated tosdr score to determine the message
+        let message = constants.tosdrMessages.unknown
+        if (tosdrData.class) {
+            message = constants.tosdrMessages[tosdrData.class]
+        } else if (matchGood.length && matchBad.length) {
+            message = constants.tosdrMessages.mixed
+        } else {
+            if (tosdrData.score < 0) {
+                message = constants.tosdrMessages.good
+            } else if (tosdrData.score === 0 && (matchGood.length || matchBad.length)) {
+                message = constants.tosdrMessages.mixed
+            } else if (tosdrData.score > 0) {
+                message = constants.tosdrMessages.bad
+            }
+        }
+
+        return {
+            score: tosdrData.score,
+            class: tosdrData.class,
+            reasons: {
+                good: matchGood,
+                bad: matchBad
+            },
+            message: message
+        }
+    }
+
+    getTosdrScore (hostname) {
+        const domain = tldjs.getDomain(hostname)
+        const parent = utils.findParent(hostname)
+
+        // grab the first available val
+        // starting with most general first
+        const score = [
+            tosdrScores[parent],
+            tosdrScores[domain],
+            tosdrScores[hostname]
```

Comment:
Add comment about possible edge case: what if `foo.bar.com` and `bar.com` are both in TOSDR and have different privacy scores?